lib/usecase/login_use_case.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:async';
+
+import 'package:injectable/injectable.dart';
+import 'package:survey/api/repository/user_repository.dart';
+import 'package:survey/api/response/login_response.dart';
+import 'package:survey/usecase/base/base_use_case.dart';
+import 'package:survey/utils/shared_preferences_utils.dart';
+
+class LoginInput {
+  final String email;
+  final String password;
+
+  LoginInput({
+    required this.email,
+    required this.password,
+  });
+}
+
+@Injectable()
+class LoginUseCase extends UseCase<void, LoginInput> {
+  final UserRepository _repository;
+  final SharedPreferencesUtils _sharedPreferencesUtils;
+
+  const LoginUseCase(
+    this._repository,
+    this._sharedPreferencesUtils,
+  );
+
+  @override
+  Future<Result<void>> call(LoginInput input) {
+    return _repository
+        .login(email: input.email, password: input.password)
+        .then((value) => _saveRefreshToken(value))
+        .onError<Exception>(
+            (exception, stackTrace) => Failed(UseCaseException(exception)));
+  }
+
+  Result<void> _saveRefreshToken(LoginResponse loginResponse) {
```

Comment:
It's not only saving refresh token, how about

```suggestion
  Result<void> _saveTokens(LoginResponse loginResponse) {
```